ui/src/app/streams/streams.service.ts

```diff
@@ -152,6 +153,17 @@ export class StreamsService {
       .catch(this.errorHandler.handleError);
   }
 
+  metrics(streamNames?: string[]): Observable<StreamMetrics.Stream[]> {
+    const options = HttpUtils.getDefaultRequestOptions();
+    if (streamNames) {
+      const params =  new URLSearchParams('', URL_QUERY_ENCODER);
+      params.append('names', streamNames.join(','));
+      options.params = params;
+    }
+    return this.http.get('/metrics/streams', options)
+      .map(res => res.json());
```

Comment:
Already commented `StreamMetrics` class but we want to return a true class here, thus you need real pojo class and not just assume json magically have same format. Also need to catch with error handler.